pkg/reconciler/instances/istio/istio_test.go

```diff
@@ -48,14 +51,124 @@ const istioctlMockCompleteVersion = `{
 		]
 	  }`
 
-func Test_RunUninstallAction(t *testing.T) {
+	istioctlMockLatestVersion = `{
+		"clientVersion": {
+		  "version": "1.11.2",
+		  "revision": "revision",
+		  "golang_version": "go1.16.7",
+		  "status": "Clean",
+		  "tag": "1.11.2"
+		},
+		"meshVersion": [
+		  {
+			"Component": "pilot",
+			"Info": {
+			  "version": "1.12.4",
+			  "revision": "revision",
+			  "golang_version": "",
+			  "status": "Clean",
+			  "tag": "1.12.4"
+			}
+		  }
+		],
+		"dataPlaneVersion": [
+		  {
+			"ID": "id",
+			"IstioVersion": "1.12.4"
+		  }
+		]
+	  }`
 
-	t.Run("Istio uninstall should also delete namespace", func(t *testing.T) {
+	istioctlMockTooNewVersion = `{
+		"clientVersion": {
+		  "version": "1.11.2",
+		  "revision": "revision",
+		  "golang_version": "go1.16.7",
+		  "status": "Clean",
+		  "tag": "1.11.2"
+		},
+		"meshVersion": [
+		  {
+			"Component": "pilot",
+			"Info": {
+			  "version": "1.13.4",
+			  "revision": "revision",
+			  "golang_version": "",
+			  "status": "Clean",
+			  "tag": "1.13.4"
+			}
+		  }
+		],
+		"dataPlaneVersion": [
+		  {
+			"ID": "id",
+			"IstioVersion": "1.13.4"
+		  }
+		]
+	  }`
+)
 
-		wsf, _ := workspace.NewFactory(nil, "./test_files", log.NewLogger(true))
+func Test_RunUpdateAction(t *testing.T) {
+	wsf, _ := workspace.NewFactory(nil, "./test_files", log.NewLogger(true))
+	model := reconciler.Task{
+		Component: "istio-configuration",
+		Namespace: "istio-system",
+		Version:   "1.11.2",
+		Profile:   "production",
+	}
+	actionContext := newActionContext(wsf, model)
 
-		actionContext := newActionContext(wsf)
+	t.Run("Istio update should permit one minor downgrade", func(t *testing.T) {
+		// given
+		providerMock := clientsetmocks.Provider{}
+		providerMock.On("RetrieveFrom", mock.Anything, mock.Anything).Return(fake.NewSimpleClientset(), nil)
+		commanderMock := commandermocks.Commander{}
+		commanderMock.On("Version", mock.Anything, mock.Anything).Return([]byte(istioctlMockLatestVersion), nil)
+		commanderMock.On("Upgrade", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+		proxy := proxymocks.IstioProxyReset{}
+		proxy.On("Run", mock.Anything).Return(nil)
+		performer := actions.NewDefaultIstioPerformer(&commanderMock, &proxy, &providerMock)
+		action := istio.NewReconcileAction(performer)
 
+		// when
+		err := action.Run(actionContext)
+
+		// then
+		require.NoError(t, err)
+		commanderMock.AssertCalled(t, "Version", mock.Anything, mock.Anything)
+		commanderMock.AssertCalled(t, "Upgrade", mock.Anything, mock.Anything, mock.Anything)
+	})
+
+	t.Run("Istio update should NOT permit more than one minor downgrade", func(t *testing.T) {
```

Comment:
Why do you uppercase `NOT` in this context?

Reply:
I think it explicitly states why this test is in place. And this pattern is consistent also across our other components. For example: https://github.com/kyma-incubator/api-gateway/blob/7e5551484fef46333cf66058fb0f6a68ec00cf8f/internal/validation/validate_test.go#L164